Error tooltip

andreasphil opened the pull request:
Implements RISDEV-3800:

- Callouts can be dismissed
- Tooltip component for showing small floating callouts next to components
- Tooltips integrated into the button

andreasphil committed:
feat: make callout dismissable

RISDEV-3800

andreasphil committed:
feat: implement tooltip component

RISDEV-3800

andreasphil committed:
fix: mark props with defaults as optional

RISDEV-3800

andreasphil committed:
test: missing unit tests

RISDEV-3800

andreasphil committed:
feat: make tooltip work as a standalone component

- Will be positioned in a floating state based on the child component
  if it has one
- Will be positioned like a regular element if there is no child
  component

RISDEV-3800

andreasphil committed:
feat: composable for getting element IDs

This can be used as an identifier for HTML elements, for example for
creating associations between different elements via ID for accessibility
purposes such as aria-labelledby or aria-describedby.

RISDEV-3800

andreasphil committed:
feat: add aria description to tooltips for embedded element

RISDEV-3800

malte-laukoetter commented on frontend/src/composables/useElementId.ts:
```diff
@@ -0,0 +1,33 @@
+   */
+  const currentIds = new Set<string>()
+
+  function getIdentifier() {
```
I'm quite unsure how good it is that the id now starts with a number and includes a `.` as both is (while technically supported) not recommended as there are a couple of pitfalls that one could fall into with them: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id

andreasphil replied:
It doesn't contain dots (those are "multiplied away") but good catch about the numbers. I'll fix that!

andreasphil committed:
feat: add prefix to element identifiers

This can make the identifiers more readable and avoids some issues with
using numbers as element IDs in HTML, see:

https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id

RISDEV-3800

andreasphil committed:
test: ensure tooltip describes the element using the tooltip

RISDEV-3800

andreasphil committed:
feat: use custom ID prefix in tooltip

RISDEV-3800
